src/main/java/net/dries007/tfc/world/feature/vein/DikeVeinFeature.java

```diff
@@ -0,0 +1,36 @@
+package net.dries007.tfc.world.feature.vein;
+
+import java.util.Random;
+import javax.annotation.Nullable;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MutableBoundingBox;
+import net.minecraft.world.ISeedReader;
+
+import com.mojang.serialization.Codec;
+import net.dries007.tfc.common.types.Rock;
+import net.dries007.tfc.world.chunkdata.ChunkData;
+import net.dries007.tfc.world.chunkdata.ChunkDataProvider;
+
+public class DikeVeinFeature extends PipeVeinFeature
```

Comment:
Basically I don't see why this needs to be a thing at all. My vision of dikes was for them to just unrelated rocks - not necessarily the bottom rock, and this hardcoding in any case is rather gross.